generic/tests/mac_change.py

```diff
@@ -18,11 +18,10 @@ def check_guest_mac(mac, vm, device_id=None):
         err = "Could not find device '%s' from query-network monitor command."
         err += "query-network command output: %s" % str(network_info)
         raise error.TestFail(err)
-    for info in str(network_info).splitlines():
-        if re.match(device_id, info.strip(), re.I) and mac not in info:
-            err = "Cold not get correct mac from qmp command!"
-            err += "query-network command output: %s" % str(network_info)
-            raise error.TestFail(err)
+    if re.match(device_id, str(network_info).strip(), re.I) and mac not in str(network_info):
```

Comment:
on the line 13： network_info = str(vm.monitor.info("network"))
`if device_id in network_info and and mac not in network_info` will do the same thing.
and the rest of `str(network_info)` can be replaced with `network_info`